test/Microsoft.AspNet.Mvc.FunctionalTests/TagHelperSampleTest.cs

```diff
@@ -18,44 +23,172 @@ public TagHelperSampleTest(MvcTestFixture<TagHelperSample.Web.Startup> fixture)
 
         public HttpClient Client { get; }
 
-        public static TheoryData<string> PathData
+        [Fact]
+        public async Task HomeController_Index_ReturnsExpectedContent()
+        {
+            // Arrange
+            var expectedMediaType = MediaTypeHeaderValue.Parse("text/html; charset=utf-8");
+            var outputFile = "compiler/resources/TagHelperSample.Web.Home.Index.html";
+            var resourceAssembly = typeof(TagHelperSampleTest).GetTypeInfo().Assembly;
+            var expectedContent = await ResourceFile.ReadResourceAsync(resourceAssembly, outputFile, sourceFile: false);
+
+            // Act
+            var response = await Client.GetAsync("http://localhost/");
+            var responseContent = await response.Content.ReadAsStringAsync();
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            Assert.Equal(expectedMediaType, response.Content.Headers.ContentType);
+
+#if GENERATE_BASELINES
+            ResourceFile.UpdateFile(resourceAssembly, outputFile, expectedContent, responseContent);
+#else
+            Assert.Equal(expectedContent, responseContent, ignoreLineEndingDifferences: true);
+#endif
+        }
+
+
+        [Fact]
+        public async Task HomeController_Create_Get_ReturnsSuccess()
+        {
+            // Act
+            var response = await Client.GetAsync("http://localhost/Home/Create");
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+        }
+
+        [Fact]
+        public async Task HomeController_Create_Post_ReturnsSuccess()
+        {
+            // Arrange
+            var createBillyContent = CreateUserFormContent("Billy", "2000-11-30", 0, "hello");
+
+            // Act
+            var response = await Client.PostAsync("http://localhost/Home/Create", createBillyContent);
+
+            // Assert
+            AssertRedirectsToHome(response);
+        }
+
+        [Fact]
+        public async Task HomeController_Edit_Get_ReturnsSuccess()
+        {
+            // Arrange
+            var createBillyContent = CreateUserFormContent("Billy", "2000-11-30", 0, "hello");
+
+            // Act
+            var createBilly = await Client.PostAsync("http://localhost/Home/Create", createBillyContent);
+            var response = await Client.GetAsync("http://localhost/Home/Edit/0");
```

Comment:
https://github.com/aspnet/Mvc/pull/3879#discussion_r49233903 but even more so here. Not gonna swap to that style for 5 lines of code.